### ISSUE TITLE
Configure production VZ graphql-engine to detect task health

### ISSUE THREAD

**Issue by frankhereford** [cityofaustin/atd-data-tech#11751]
Currently, the VZ graphql-engine configuration in ECS does not have a method to check on the health of the running tasks, and this is shown in the AWS console as "Health Status" being `Unknown`. The configuration should be enhanced to check, via `GET`, the [healthz](https://hasura.io/docs/latest/api-reference/health/) API call. When this is complete, future updates to the task definition will be able to be deployed without manually killing old instances, as the system is able to tell when a task comes up and is ready to serve requests.

**Comment by johnclary** [comment]
@frankhereford i'm struggling to find the commit or where i commented about this, but i added this healthcheck to the task definition for `moped-test-graphql-endpoint` :

```json
            "healthCheck": {
                "command": [
                    "CMD-SHELL",
                    "timeout 1s bash -c ':> /dev/tcp/127.0.0.1/8080' || exit 1"
                ],
                "interval": 5,
                "timeout": 2,
                "retries": 10
            }
```

i'm forgetting exactly why i had to go this route vs the out-of-the-box `healthz` endpoint. I will try to track that down but wanted to share before i forget.

**Comment by johnclary** [comment]
edit: @frankhereford the original issue with the healthcheck is that `curl` was not available inside hasura's default images. fortunately that apepars to have been fixed in `v2.18` ([thread](https://github.com/hasura/graphql-engine/issues/1532#issuecomment-1410208460)).

TLDR: I'm in full support of doing this the right way using Hasura's built-in healthcheck.

**Comment by frankhereford** [comment]
@johnclary Maybe you and I can sync real quick tomorrow. I may be confused -- I was thinking that we'd configure ECS to use its internal (to ECS) curl to pull the healthz path running on the containers to check their health. My goal is to get ECS aware of when a graphql-engine instance hangs so it can fire up a replacement and reap the dead one.
